packages/frontend-2/pages/workspaces/[id]/index.vue

```diff
@@ -4,12 +4,37 @@
   </div>
 </template>
 <script setup lang="ts">
+import { useQuery } from '@vue/apollo-composable'
 import { useRoute } from 'vue-router'
+import { useWorkspacesMixpanel } from '~/lib/workspaces/composables/mixpanel'
+import { workspaceMixpanelUpdateGroupQuery } from '~/lib/workspaces/graphql/queries'
 
 const route = useRoute()
 const workspaceId = computed(() => route.params.id as string)
+const { workspaceMixpanelUpdateGroup } = useWorkspacesMixpanel()
+const pageFetchPolicy = usePageQueryStandardFetchPolicy()
 
 definePageMeta({
   middleware: ['requires-workspaces-enabled', 'require-valid-workspace']
 })
+
+const { result } = useQuery(
+  workspaceMixpanelUpdateGroupQuery,
```

Comment:
why not merge it with the main workspace page query that already exists in WorkspaceProjectList? this will unnecessarily lengthen page load times, especially since its quite a hefty query

Reply:
I separated them so they get too intertwined, if subsequent queries fetch the same data they will leverage the cache or not?

Reply:
These both run at the same time, so there's nothing in the cache and they end up executing the same thing twice. Leveraging the cache would only be relevant if the data were already in the cache when the 2nd query starts, but here they both run concurrently.

Generally, for any kind of page load queries (not some query triggered async on button press or whatever) you just want everything to be loaded with as few queries as possible, because the cache is gonna be empty at that time (in SSR).

Reply:
Good point, I've updated it to be part of the same query